test/parallel/test-async-wrap-asyncresource-constructor.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+require('../common');
+
+// This tests that AsyncResource throws an error if bad parameters are passed
+
+const assert = require('assert');
+const AsyncResource = require('async_hooks').AsyncResource;
+const async_wrap = process.binding('async_wrap');
+
+assert.throws(() => {
+  return new AsyncResource();
+}, new RegExp('^TypeError: type must be a string with length > 0$'));
```

Comment:
I am not sure @jasnell’s nit was actually addressed… why are these `new RegExp()` calls? It might be confusing to readers of our tests because we almost exclusively use them to create *dynamic* regexes that depend on variables.